### PR TITLE
cloud-1237 add datavirt tests

### PR DESCRIPTION
**Pull request by bdecoste**

### CODE REVIEW

File: datavirt/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#Arquillian Tests for Red Hat Single-Sign On (SSO)
+This project contains OpenShift v3 tests for the Red Hat SSO/Keycloak image
+and the SSO/Keycloak capabilities of the EAP image
+
+##Requirements
+The following are required to run the tests:
+ * ImageStreams currently pull the SSO and SSO-enabled EAP images from docker-registry.usersys.redhat.com so this registry must be enabled for the Docker service
```
Comment (rimolive) [inline datavirt/README.md]: All text is referencing SSO, not Datavirt

File: datavirt/pom.xml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-surefire-plugin</artifactId>
+				<configuration>
+					<runOrder>alphabetical</runOrder>
+					<redirectTestOutputToFile>false</redirectTestOutputToFile>
```
Comment (rimolive) [inline datavirt/pom.xml]: Change attribute redirectTestOutputToFile to true. It's how all other tests are configured.

File: datavirt/src/test/resources/arquillian.xml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    <!--
+    <engine>
+	         <property name="deploymentExportPath">target/arquillian</property>
+    </engine>
+    -->
+    <!-- Force the use of the Servlet 3.0 protocol with all containers, as it is the most mature -->
+    <defaultProtocol type="jmx-as7" />
+    
+    <extension qualifier="ce-cube">
+        <property name="kubernetesNamespacePrefix">cearq-sso</property>
```
Comment (rimolive) [inline datavirt/src/test/resources/arquillian.xml]: Change to 'cearq-datavirt'

File: datavirt/src/test/java/org/jboss/test/arquillian/ce/datavirt/DatavirtBasicTest.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.test.arquillian.ce.datavirt.support.JDBCClient;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+@Template(url = "https://raw.githubusercontent.com/jboss-openshift/application-templates/master/datavirt/datavirt63-basic-s2i.json",
```
Comment (jwendell) [inline datavirt/src/test/java/org/jboss/test/arquillian/ce/datavirt/DatavirtBasicTest.java]: can you use the ${template.repository...} and ${template.branch...} instead of hard coding it?

File: datavirt/src/test/java/org/jboss/test/arquillian/ce/datavirt/DatavirtLogTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.test.arquillian.ce.datavirt.support.JDBCClient;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+@Template(url = "https://raw.githubusercontent.com/jboss-openshift/application-templates/master/datavirt/datavirt63-basic-s2i.json",
```
Comment (jwendell) [inline datavirt/src/test/java/org/jboss/test/arquillian/ce/datavirt/DatavirtLogTest.java]: can you use the ${template.repository...} and ${template.branch...} instead of hard coding it?

File: datavirt/src/test/java/org/jboss/test/arquillian/ce/datavirt/DatavirtSecureTest.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.test.arquillian.ce.datavirt.support.JDBCClient;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+@Template(url = "https://raw.githubusercontent.com/bdecoste/application-templates/vdbtests/datavirt/datavirt63-secure-s2i.json",
```
Comment (jwendell) [inline datavirt/src/test/java/org/jboss/test/arquillian/ce/datavirt/DatavirtSecureTest.java]: does this template exist on our repo? if not, can you get it merged there and then update this line to point to it (using ${template.repository...} and ${template.branch...})

File: datavirt/src/test/resources/arquillian.xml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      class mode causes the test pod to be recreated for each test class. this
+      is required to pickup environment variables specific to the test
+      environment, e.g. service host, port, etc.  testrunner relates to the pod
+      name in testrunner-pod.json.
+    -->
+    <container qualifier="testrunner" mode="class" default="true">
+        <!-- Pod running remote tests. -->
+        <configuration>
+            <!-- For use with archillian-chameleon -->
+            <property name="target">jboss eap:6.4.5:remote</property>
```
Comment (jwendell) [inline datavirt/src/test/resources/arquillian.xml]: This line is not necessary, please remove it.

File: datavirt/src/test/resources/testrunner-pod.json [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+	"apiVersion":"v1",
+	"kind": "Pod",
+	"metadata": {
+		"name": "testrunner",
+		"labels": {
+			"name": "testrunner"
+		}
+	},
+	"spec": {
+	    "terminationGracePeriodSeconds": 300,
```
Comment (jwendell) [inline datavirt/src/test/resources/testrunner-pod.json]: This should be set to 0.

File: datavirt/src/test/resources/testrunner-pod.json [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+		"containers": [{
+			"name": "testrunner",
+			"readinessProbe" : {
+				"exec": {
+					"command": [
+					    "/bin/bash",
+					    "-c",
+					    "curl -s --digest -L http://localhost:9990/management --header \"Content-Type: application/json\" -d '{\"operation\":\"read-attribute\",\"name\":\"server-state\",\"json.pretty\":1}' | grep -iq running"
```
Comment (jwendell) [inline datavirt/src/test/resources/testrunner-pod.json]: Use "/opt/eap/bin/readinessProbe.sh" instead

File: datavirt/src/test/resources/testrunner-pod.json [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+					]
+				}
+			},
+			"image": "jboss-eap-6/eap64-openshift:1.2",
```
Comment (jwendell) [inline datavirt/src/test/resources/testrunner-pod.json]: We are using latest image, 1.4

File: datavirt/src/test/resources/testrunner-pod.json [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+                {
+                    "name": "MQ_QUEUES",
+                    "value": "KIE.SERVER.REQUEST,KIE.SERVER.RESPONSE"
+                },
+                {
+                    "name": "MQ_TOPICS",
+                    "value": ""
+                },
+                {
+                	"name": "ADMIN_USERNAME",
```
Comment (jwendell) [inline datavirt/src/test/resources/testrunner-pod.json]: Are those MQ_* variables really needed?

File: datavirt/src/test/resources/web.xml [hunk 12]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jwendell) [inline datavirt/src/test/resources/web.xml]: Is this file really needed?

**Comment by bdecoste** [comment]
I need to use the vdbtests branch of the bdecoste repo until https://github.com/jboss-openshift/application-templates/pull/217 is merged

File: datavirt/src/test/java/org/jboss/test/arquillian/ce/datavirt/DatavirtSecureTest.java (hunk 6 above)
Comment (bdecoste) [inline datavirt/src/test/java/org/jboss/test/arquillian/ce/datavirt/DatavirtSecureTest.java]: I need to use the vdbtests branch of the bdecoste repo for DatavirtSecureTest until jboss-openshift/application-templates#217 is merged

Comment (jwendell) [inline datavirt/src/test/java/org/jboss/test/arquillian/ce/datavirt/DatavirtSecureTest.java]: @bdecoste it seems it got merged \o/
